datalad/support/sshconnector.py

```diff
@@ -267,16 +274,17 @@ def open(self):
         # wait till the command exits, connection is conclusively
         # open or not at this point
         exit_code = proc.wait()
-        ret = exit_code == 0
 
-        if not ret:
-            lgr.warning(
-                "Failed to run cmd %s. Exit code=%s\nstdout: %s\nstderr: %s",
-                cmd, exit_code, stdout, stderr
+        if exit_code != 0:
+            raise CommandError(
+                str(cmd),
+                'Failed to open SSH connection (could not start ControlMaster process)',
```

Comment:
What about defining a dedicated exception subclassed from CommandError? With adding a debug level message with details and custom str for exception we could avoid dumping into user's face that long cryptic invocation line too

Reply:
Considering how `CommandError` makes it too often into the user's face, we might want to change `CommandError` instead, anyway. We could change its `__str__` to only consider `self.msg` and make the former `__str__` a `details` property, that can be accessed by debug level log.

Reply:
I RF'ed to use a dedicated exception. I agree that further optimization along the suggested lines should be done in a generic fashion and in a separate PR.